Detectors/GlobalTracking/include/GlobalTracking/MatchTOFBase.h

```diff
@@ -0,0 +1,305 @@
+  template <typename V>
+  void makeConstrainedTPCTracks(V& container)
+  {
+    checkRefitter();
+    int nmatched = mMatchedTracks[o2::dataformats::MatchInfoTOF::TrackType::TPC].size(), nconstrained = 0;
+    container.resize(nmatched);
+    for (unsigned i = 0; i < nmatched; i++) {
+      if (makeConstrainedTPCTrack(i, container[nconstrained])) {
+        nconstrained++;
+      }
+    }
+    container.resize(nconstrained);
+  }
+
+ protected:
+  virtual bool prepareTracks();
+  virtual bool prepareTPCTracks();
+  virtual bool prepareTOFClusters();
+
+  void doMatching(int sec, o2::dataformats::MatchInfoTOF::TrackType type = o2::dataformats::MatchInfoTOF::TrackType::ITSTPC);
+  void doMatchingForTPC(int sec);
+  void selectBestMatches();
+  void selectBestMatchesHP();
+  bool propagateToRefX(o2::track::TrackParCov& trc, float xRef /*in cm*/, float stepInCm /*in cm*/, o2::track::TrackLTIntegral& intLT);
+  bool propagateToRefXWithoutCov(o2::track::TrackParCov& trc, float xRef /*in cm*/, float stepInCm /*in cm*/, float bz);
+
+  void updateTimeDependentParams();
+  void checkRefitter();
+  bool makeConstrainedTPCTrack(int matchedID, o2::dataformats::TrackTPCTOF& trConstr);
+
+  //================================================================
+
+  // Data members
+
+  // for derived class
+  int mCurrTracksTreeEntry = 0;  ///< current tracks tree entry loaded to memory
+  bool mSAInitDone = false;      ///< flag that standalone init already done
+  bool mWFInputAttached = false; ///< flag that the standalone input is attached
+  Bool_t mIsworkflowON = kFALSE; ///< flag used in the standalone mode
+
+  float mXRef = Geo::RMIN; ///< reference radius to propage tracks for matching
+
+  bool mMCTruthON = false; ///< flag availability of MC truth
+
+  ///========== Parameters to be set externally, e.g. from CCDB ====================
+  float mBz = 0; ///< nominal Bz
+
+  // to be done later
+  float mTPCTBinMUS = 0.;    ///< TPC time bin duration in microseconds
+  float mTPCTBinMUSInv = 0.; ///< inverse TPC time bin duration in microseconds
+  float mTPCBin2Z = 0.;      ///< conversion coeff from TPC time-bin to Z
+
+  bool mIsCosmics = false;    ///< switch on to reconstruct cosmics and match with TPC
+  float mTimeTolerance = 1e3; ///< tolerance in ns for track-TOF time bracket matching
+  float mSpaceTolerance = 10; ///< tolerance in cm for track-TOF time bracket matching
+  int mSigmaTimeCut = 30.;    ///< number of sigmas to cut on time when matching the track to the TOF cluster
+
+  bool mIsITSTPCused = false;
+  bool mIsTPCused = false;
+  bool mIsTPCTRDused = false;
+  bool mIsITSTPCTRDused = false;
+  bool mSetHighPurity = false;
+
+  ///>>>------ these are input arrays which should not be modified by the matching code
+  //           since this info is provided by external device
+  gsl::span<const o2::dataformats::TrackTPCITS> mITSTPCTracksArrayInp;  ///< input tracks
```

Comment:
also, by giving up non-DPL version you can also get rid of all these spans, since they will be available from the RecoContainer.